inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -215,68 +212,102 @@ def arxiv_plot_extract(obj, eng):
-                    except XMLSyntaxError:
-                        # Probably the %auto-ignore comment exists, so we skip the
-                        # first line. See: inspirehep/inspire-next/issues/2195
-                        authors_xml = convert(
-                            xml_content.split('\n', 1)[1],
-                            stylesheet,
-                        )
-
-                    extracted_authors.extend(marcxml2record(authors_xml).get('authors', []))
-
-            if extracted_authors:
-                for author in extracted_authors:
-                    author['full_name'] = decode_latex(author['full_name'])
-
-                obj.data['authors'] = extracted_authors
-                set_mark(obj, "authors_xml", True)
+    # Probably the %auto-ignore comment exists, so we skip the
+    # first line. See: inspirehep/inspire-next/issues/2195
+    if "%auto-ignore" in xml_content:
+        xml_content = xml_content.split('\n', 1)[1]
+
+    content = Selector(text=xml_content, type="xml")
+
+    content.remove_namespaces()
+    # Going through all the authors in the file
+    for author in content.xpath("//Person"):
+
+        ids = []
+        affiliations = []
+
+        # Getting all the author ids
+        for source, id in itertools.izip(author.xpath('./authorIDs/authorID[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/@source | ./authorids/authorid[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/@source').getall(), author.xpath('./authorIDs/authorID[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/text() | ./authorids/authorid[@source!="undefined" and @source!="UNDEFINED" and @source!="INSPIRE-" and @source!="inspire-" and @source!="" and text()!="" and text()!="undefined" and text()!="UNDEFINED"]/text()').getall()):
+            if source == u'CCID':
+                ids.append(['CERN', id])
+            elif source == u'INSPIRE':
+                ids.append(['{} ID'.format(source), id])
             else:
-                set_mark(obj, "authors_xml", False)
+                ids.append([source, id])
 
-    return _author_list
+        # Getting all the names for affiliated organizations using the organization ids from author
+        for affiliation in author.xpath("./authorAffiliations/authorAffiliation/@organizationid").getall():
+            affiliations.append(content.xpath('//organizations/Organization[@id="' + affiliation + '"]/orgName[@source="spiresICN" or @source="INSPIRE" or @source="ROR" and text()!="undefined" and text()!="UNDEFINED" and text()!="" ]/text()').get())
```

Comment:
* are you putting the ROR in the right place? in the end, it should end up in [`affiiliations_identifiers`](https://github.com/inspirehep/inspire-schemas/blob/270dbb0d25a8d652b669a3456d6ef02b29e0f824/inspire_schemas/records/hep.yml#L294-L320) in the record, not `affiliations` (see issue for test case).
* you shouldn't do string formatting with `+` (see previous comment).